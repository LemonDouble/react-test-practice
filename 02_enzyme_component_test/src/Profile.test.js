import React from "react";
import { mount } from "enzyme";
import Profile from "./Profile";

describe("<Profile />", () => {
  it("matches snapShot", () => {
    const wrapper = mount(<Profile username="lemonDouble" name="가나다" />);
    expect(wrapper).toMatchSnapshot();
  });

  it("renders username and name", () => {
    const wrapper = mount(<Profile username="lemonDouble" name="가나다" />);
    expect(wrapper.props().username).toBe("lemonDouble");
    expect(wrapper.props().name).toBe("가나다");

    const boldElement = wrapper.find("b");
    expect(boldElement.contains("lemonDouble")).toBe(true);

    const spanElement = wrapper.find("span");
    expect(spanElement.contains("가나다")).toBe(true);
  });
});
