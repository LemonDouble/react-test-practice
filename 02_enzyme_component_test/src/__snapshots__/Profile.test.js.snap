// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> matches snapShot 1`] = `
<Profile
  name="가나다"
  username="lemonDouble"
>
  <div>
    <b>
      lemonDouble
    </b>
     
    <span>
      (
      가나다
      )
    </span>
  </div>
</Profile>
`;
